### PR TITLE
Improvements to make 4.x compatible with different projects

### LINKED ISSUE #5125: Improvements to make 4.x compatible with different projects
While migrating [truffle](https://github.com/trufflesuite/truffle) came across a few changes which are required to make web3@4x compatible with a lot of projects our there. 

Here is the the list of individual changes: 

- [x] Need to change the TS target to ES2016 for broader compatibility 
- [x] Need to export `formatters` as namespace from `web3-common` package. 
- [x] `web3.eth.net` namespace is not exported 
- [x] `web3.eth.personal` namespace is not exported 
- [x] `web3.utils` namespace is not exported
- [x] `web3.modules` namespace is not exported
- [x] `Web3.utils` static namespace is not exported 
- [x] `Web3.version` static namespace is not exported 
- [x] Parameter for the web3 constructor is required, need to be optional for 1.x compatibility 
- [x] Parameter for web3.setProvider is required, need to be optional for 1.x compatibility. 
- [x] Truffle `HDWalletProvider` type is not assignable to `SupportedProviders` type. 
- [x] Would be nice to rename `PromiEvent` class to `Web3PromiEvent` to make it more compatible with 1.x implementation. 
- [x] Every provider should have have `send` method for backward compatibility.

### PR DESCRIPTION
**Pull request by nazarhussain**
## Description

Fixes #5125

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` with success.
- [x] I have tested the built `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Review by avkos — APPROVED** [review]

**Review by jdevcs — CHANGES_REQUESTED** [review]
Some of tests are failing and needs to be fixed:

```
Cannot find name 'Web3PromiEvent'.

```

File: packages/web3-common/src/web3_base_provider.ts [hunk 1]
```diff
@@ -51,6 +51,27 @@ export abstract class Web3BaseProvider<API extends Web3APISpec = EthExecutionAPI
 	public abstract getStatus(): Web3BaseProviderStatus;
 	public abstract supportsSubscriptions(): boolean;
 
+	/**
+	 * @deprecated Please use `.request` instead.
+	 *
+	 * @param payload - Request Payload
+	 * @param cb - Callback
+	 */
+	public send<Method extends Web3APIMethod<API>, ResponseType = Web3APIReturnType<API, Method>>(
```
Comment (jdevcs) [inline packages/web3-common/src/web3_base_provider.ts:61]: create `sendAsync` as well for supporting legacy apps https://eips.ethereum.org/EIPS/eip-1193#sendasync-deprecated

File: packages/web3-common/src/web3_base_provider.ts [hunk 2]
```diff
@@ -51,6 +51,27 @@ export abstract class Web3BaseProvider<API extends Web3APISpec = EthExecutionAPI
...
+		payload: Web3APIPayload<API, Method>,
+		// Used "null" value to match the legacy version
+		// eslint-disable-next-line @typescript-eslint/ban-types
+		cb: (err?: Error | null, response?: JsonRpcResponse<ResponseType>) => void,
```
Comment (jdevcs) [inline packages/web3-common/src/web3_base_provider.ts]:
```suggestion
		cb: (err?: Error | Web3Error | null, response?: JsonRpcResponse<ResponseType>) => void,
```

File: packages/web3-common/src/web3_base_provider.ts [hunk 3]
```diff
@@ -51,6 +51,27 @@ export abstract class Web3BaseProvider<API extends Web3APISpec = EthExecutionAPI
...
+	) {
+		this.request(payload)
+			.then(response => {
+				cb(undefined, response);
+			})
+			.catch((err: Error) => {
```
Comment (jdevcs) [inline packages/web3-common/src/web3_base_provider.ts]:
```suggestion
			.catch((err: Error | Web3Error ) => {
```

File: packages/web3-eth-iban/src/web3_iban.ts [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { toChecksumAddress, isAddress, leftPad, hexToNumber, HexString } from 'web3-utils';
+import { InvalidAddressError } from 'web3-errors';
+import { IbanOptions } from './types';
+
+export class Web3Iban {
```
Comment (jdevcs) [inline packages/web3-eth-iban/src/web3_iban.ts]:
```suggestion
export class Iban {
```

Make it compatible to 1.x

File: packages/web3-eth-iban/src/index.ts [hunk 5]
```diff
@@ -15,218 +15,9 @@ You should have received a copy of the GNU Lesser General Public License
...
-	}
-
-	/**
-	 * check if iban number if indirect
-	 */
-	public isIndirect(): boolean {
-		return Iban.isIndirect(this._iban);
-	}
-}
+export default Web3Iban;
```
Comment (jdevcs) [inline packages/web3-eth-iban/src/index.ts]:
```suggestion
export default Iban;
```
Make it compatible to 1.x

File: packages/web3-eth-iban/src/index.ts [hunk 6]
```diff
@@ -15,218 +15,9 @@ You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { toChecksumAddress, isAddress, leftPad, hexToNumber, HexString } from 'web3-utils';
-import { InvalidAddressError } from 'web3-errors';
-import { IbanOptions } from './types';
+import { Web3Iban } from './web3_iban';
```
Comment (jdevcs) [inline packages/web3-eth-iban/src/index.ts]:
```suggestion
import { Iban } from './web3_iban';
```
Make it compatible to 1.x

File: packages/web3-eth-iban/src/index.ts [hunk 7]
```diff
@@ -15,218 +15,9 @@ You should have received a copy of the GNU Lesser General Public License
...
 
-export class Iban {
```
Comment (jdevcs) [inline packages/web3-eth-iban/src/index.ts:22]: Why we need this refactor to new file?

File: packages/web3-eth-iban/test/unit/iban.test.ts [hunk 8]
```diff
@@ -15,7 +15,7 @@ You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { Iban } from '../../src/index';
+import { Web3Iban } from '../../src/web3_iban';
```
Comment (jdevcs) [inline packages/web3-eth-iban/test/unit/iban.test.ts]: All changes in this file should be reverted as its more good to use `Iban` name for making it more compatible to 1.x

File: packages/web3-eth-personal/src/eth_personal.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { Web3Context } from 'web3-core';
```
Comment (jdevcs) [inline packages/web3-eth-personal/src/eth_personal.ts]: Why we need this refactor?

**Review by jdevcs — CHANGES_REQUESTED** [review]

File: packages/web3-common/src/web3_base_provider.ts (hunk 1 above)
Comment (nazarhussain) [reply packages/web3-common/src/web3_base_provider.ts:61]: Done.

File: packages/web3-eth-iban/src/index.ts (hunk 7 above)
Comment (nazarhussain) [reply packages/web3-eth-iban/src/index.ts:22]: To have a consistent pattern around the code base. To have classes in files as their names. It's easy to search in the code.

**Review by luu-alex — APPROVED** [review]

**Review by nikoulai — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

**Pull request merged**